app/src/main/java/com/example/cristiana/weatherapp/TenDaysActivity.java

```diff
@@ -0,0 +1,212 @@
+
+import com.example.cristiana.weatherapp.model.Forecast;
+import com.example.cristiana.weatherapp.model.TenDaysWeather;
+import com.example.cristiana.weatherapp.model.WeatherApp;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Created by Cristiana on 4/28/2017.
+ */
+
+public class TenDaysActivity extends AppCompatActivity {
+    RecyclerView mRecyclerView;
+    Adapter adapter;
+    boolean mLandscape;
+
+    public static final String LOC_SAVED = "loc_saved";
+    String mLocationName;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_tendays);
+
+        /* get the location name */
+        mLocationName = getIntent().getStringExtra(LOC_SAVED);
+        fetchWeather();
+
+        /* set layout manager */
+        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        /* check the orientation */
+        mLandscape = (findViewById(R.id.container) != null);
+
+        /* set adapter */
+        adapter = new Adapter(new Adapter.Callback() {
+
+            @Override
+            public void show(Forecast forecast) {
+                if(mLandscape) {
+                    Fragment details = DayDetailsFragment.New(forecast, mLocationName);
+                    getSupportFragmentManager().beginTransaction()
+                            .replace(R.id.container, details)
+                            .commit();
+                } else {
+                    Intent intent = new Intent(TenDaysActivity.this, DayDetailsActivity.class);
+                    intent.putExtra("location", mLocationName);
+                    intent.putExtra("temperature", forecast.getMain().getTemp());
+                    intent.putExtra("description", forecast.getWeather().get(0).getDescription());
+                    intent.putExtra("pressure", forecast.getMain().getPressure());
+                    intent.putExtra("wind_speed", forecast.getWind().getSpeed());
+                    startActivity(intent);
+                }
+
+            }
+        });
+        mRecyclerView.setAdapter(adapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater menuInflater = getMenuInflater();
+        menuInflater.inflate(R.menu.oneday_menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.Return) {
+            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+            preferences.edit().remove(LOC_SAVED);
```

Comment:
if you don't call commit or apply after editing shared preferences, they don't get saved...